[MRG] Fix for sphinx-gallery 0.3.1 + 404 errors on Debian Jessie packages

lesteve opened the pull request:
sphinx-gallery 0.3 has changed the output figure numbering in some edge cases (mostly 0.3.1 ignores matplotlib figure numbers see https://github.com/sphinx-gallery/sphinx-gallery/issues/464 for more details). This PR replaces #13513 and should fix the CircleCI build failures in master, e.g. circleci.com/gh/scikit-learn/scikit-learn/52538.

lesteve committed:
Fix for sphinx-gallery 0.3.1.

Figure numbering have been changed to ignore matplotlib figure number.

jnothman reviewed (approved):
Hair to give it a try if it passes locally

lesteve committed:
Use circleci/3.6 image.

This docker image uses Debian stretch and fixes the problems seen with apt-get update with Debian jessie.

lesteve commented:
I pushed a simple fix for the CircleCI problem with jessie (circleci/python3.6 uses Debian stretch rather than Debian jessie and should get rid of the problem):
```
W: Failed to fetch http://deb.debian.org/debian/dists/jessie-updates/main/binary-amd64/Packages  404  Not Found
```

thomasjpfan commented:
Running `ag -i "plt.figure\(\w+\)" examples` on https://github.com/scikit-learn/scikit-learn/pull/13527/commits/ad6a28bdd634cd85eac2d6ae97330d99ffcd5591  returns:
```bash
examples/ensemble/plot_feature_transformation.py
94:plt.figure(1)
107:plt.figure(2)

examples/cluster/plot_affinity_propagation.py
49:plt.figure(1)

examples/cluster/plot_color_quantization.py
85:plt.figure(1)
91:plt.figure(2)
97:plt.figure(3)

examples/cluster/plot_adjusted_for_chance_measures.py
74:plt.figure(1)
104:plt.figure(2)

examples/cluster/plot_kmeans_digits.py
108:plt.figure(1)

examples/cluster/plot_mean_shift.py
45:plt.figure(1)

examples/gaussian_process/plot_gpc_isoprobability.py
67:fig = plt.figure(1)

examples/linear_model/plot_lasso_coordinate_descent_path.py
49:plt.figure(1)
64:plt.figure(2)
77:plt.figure(3)

examples/feature_selection/plot_feature_selection.py
43:plt.figure(1)

examples/manifold/plot_mds.py
61:fig = plt.figure(1)

examples/exercises/plot_iris_exercise.py
42:    plt.figure(fig_num)

examples/applications/plot_outlier_detection_housing.py
80:    plt.figure(1)
86:    plt.figure(2)
97:plt.figure(1)  # two clusters
119:plt.figure(2)  # "banana" shape
```

lesteve committed:
Install additional fonts.

Seems to be needed to convert doc/images/iris.svg.

adrinjalali commented:
curious, how did you find which fonts it needs?

adrinjalali commented:
Could you please also do a `[doc build]` commit to check the complete build?

lesteve committed:
Another missed example using figure number 0.

[doc build]

lesteve commented:
@thomasjpfan basically the problem with sphinx-gallery 0.3.1 only appears when your use figure numbers that don't match the default matplotlib figure numbering (i.e. 1, 2, 3, ..., `n_figures`). So I only fixed the examples with non-default matplotlib numbering (I even cheated intially and did `git grep 000.png` to find the included images that were starting at 000.png rather than 001.png). After that I did look at `git grep -P 'plt.figure\(.+\)' to figure out which additional examples were problematic.

@adrinjalali I used the using `docker` with the `circleci/python3.6` image locally to quickly debug. I then guessed by trial and error + googling. Luckily (or unluckily it depends) I had a similar convert error recently on my computer (for some reasons /etc/ImageMagick-6/policy.xml was disabling PS conversions ...)

I pushed a commit fixing an example I missed before and `[doc build]`.

adrinjalali reviewed (approved):
Awesome!

lesteve commented:
FYI the CircleCI is green on master too: https://circleci.com/gh/scikit-learn/scikit-learn/53038 (you never know the pdf build is more finicky sometimes ...).

jnothman commented:
Thanks for fixing this everyone
